spec/models/tenant_spec.rb

```diff
@@ -186,163 +186,6 @@
-        root_tenant.use_config_for_attributes = false
-        expect(root_tenant).to be_logo
-      end
-
-      context "#with custom_logo configuration" do
-        it "knows there is a logo from configuration" do
-          stub_settings(:server => {:custom_logo => true})
-          expect(root_tenant).to be_logo
-        end
-
-        it "knows there is no logo when not using config" do
-          root_tenant.use_config_for_attributes = false
-          expect(root_tenant).not_to be_logo
-        end
-      end
-    end
-  end
-
-  # NOTE: much of this functionality was tested in #logo?
-  describe "#logo_file_name" do
-    it "has custom filename" do
-      tenant.logo_file_name = "custom_logo.png"
-      expect(tenant.logo_file_name).to eq("custom_logo.png")
-    end
-  end
-
-  describe "#logo_content_type" do
-    it "has custom content_type" do
-      tenant.logo_content_type = "image/jpg"
-      expect(tenant.logo_content_type).to eq("image/jpg")
-    end
-
-    it "has no custom content_type" do
-      expect(tenant.logo_content_type).to be_nil
-    end
-
-    context "for root_tenant" do
-      it "has custom content_type" do
-        expect(root_tenant.logo_content_type).to eq("image/png")
-      end
-
-      context "#with custom logo configuration" do
-        let(:settings) { {:server => {:custom_logo => true}} }
-
-        it "has custom content_type" do
-          expect(root_tenant.logo_content_type).to eq("image/png")
-        end
-      end
-    end
-  end
-
-  describe "#login_logo" do
-    # NOTE: initializers/paperclip.rb sets up :default_login_logo
-
-    it "has a default login image" do
-      tenant.save!
-      expect(tenant.login_logo.url).to match(/login-screen-logo.*\.png/)
-    end
-
-    it "has a default login image for root_tenant" do
-      expect(root_tenant.login_logo.url).to match(/login-screen-logo.*\.png/)
-    end
-
-    it "has custom login logo" do
-      tenant.update_attributes(:login_logo_file_name => "custom_login_logo.png")
-      expect(tenant.login_logo.url).to match(/custom_login_logo.png/)
-    end
-
-    context "with custom login logo configuration" do
-      it "has custom login logo" do
-        stub_settings(:server => {:custom_login_logo => true})
-        expect(root_tenant.login_logo.url).to match(/custom_login_logo.png/)
-      end
-    end
-  end
-
-  describe "#login_logo?" do
-    it "knows when there is a login logo" do
-      expect(described_class.new(:login_logo_file_name => "image.png")).to be_login_logo
-    end
-
-    it "knows when there is no login logo" do
-      expect(described_class.new).not_to be_login_logo
-    end
-  end
-
-  describe "#nil_blanks" do
-    it "nulls out blank domain" do
-      expect(described_class.create!(:domain => "  ", :parent => root_tenant).domain).to be_nil
-    end
-
-    it "nulls out blank subdomain" do
-      expect(described_class.create!(:subdomain => "  ", :parent => root_tenant).domain).to be_nil
-    end
-  end
```

Comment:
I think @kbrock got a little too zealous to delete... 🔥 😮  I believe this `nil_blanks` is still needed.  It doesn't seem related to the logo.